tsdb/db.go

```diff
@@ -621,6 +621,18 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 
 	var wlog *wal.WAL
 	segmentSize := wal.DefaultSegmentSize
+
+	defer func() {
```

Comment:
Shouldn't we do this immediately after `&DB`?

Reply:
I think that starting the wal is the first thing that opens a file, so it works where it is, but it does no harm doing right after `&DB`. It might be easier to read also...